### DISCUSSION TITLE
How to handle requests sent into broken socket?

### DISCUSSION THREAD

**Discussion by lukashass** [feathersjs/feathers#3332] [Q&A]
### What works

Handling sockets that are already timed out fully is easily possible. Just never send the request in the first place and show an error.

### What doesn't work

When a socket just broke but has not yet timed out, we don't know about it. When we now send the request it never finishes as feathers itself doesn't seem to have a timeout.
The request will complete when the socket is reestablished, but since the user can close the browser in-between this is not sufficient.
How to detect that the request was sent into a broken socket?

**Comment by daffl** [comment]
It looks like this can now be done in Socket.io with the [ackTimeout](https://socket.io/docs/v4/client-options/#acktimeout) setting when establishing the connection:

```ts
import { io } from "socket.io-client";

const socket = io({
  ackTimeout: 10000 // time out after 10 seconds
});
```
